python/src/utils.h

```diff
@@ -90,6 +90,17 @@ inline array to_array(
   }
 }
 
+inline void throw_if_invalid_index(const array& src, int sub_, int axis) {
```

Comment:
Not used anywhere.

Reply:
Pushed new code related to this one. Previously, if we have something like this:
```
a = mx.array([0,1,2,3])
a[[0, 10]]
```
It will not throw; but it obviously should (because the index is out of bounds for dimension 0).
I now added the required checks, using `throw_if_invalid_index` to throw appropriate exception